crates/factor-sqlite/src/lib.rs

```diff
@@ -114,13 +114,34 @@ pub struct AppState {
     get_connection_pool: host::ConnectionPoolGetter,
 }
 
-/// A store of connections for all accessible databases for an application
+/// A pool of connections for a particular SQLite database
 #[async_trait]
 pub trait ConnectionPool: Send + Sync {
     /// Get a `Connection` from the pool
     async fn get_connection(&self) -> Result<Arc<dyn Connection + 'static>, v2::Error>;
 }
 
+/// A simple [`ConnectionPool`] that always creates a new connection.
+pub struct SimpleConnectionPool(
```

Comment:
Is this wrapper needed? Could you directly `impl<F: Fn() -> ...> ConnectionPool for F`?

Reply:
Yea, I find it easier to read with the simple wrapper struct, but I don't feel strongly either way.